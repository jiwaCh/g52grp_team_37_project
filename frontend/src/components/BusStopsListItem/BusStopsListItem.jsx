import React from 'react';
import proptypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { ListItem, ListItemText, ListItemIcon } from '@material-ui/core';
import LocationIcon from '@material-ui/icons/LocationOn';
import cx from 'classnames';
import useStyles from './BusStopsListItem.styles';
import BusLineChip from '../BusLineChip';

const BusStopsListItem = ({ name, id, alternate, buses }) => {
  const classes = useStyles();

  return (
    <ListItem
      button
      component={NavLink}
      to={`/busStop?stop=${id}`}
      key={id}
      className={cx({ [classes.alternate]: alternate })}
    >
      <ListItemIcon>
        <LocationIcon />
      </ListItemIcon>
      <ListItemText
        primary={name}
        secondary={
          <span className={classes.chipRoot}>
            {buses.map((busDetails, i) => (
              <BusLineChip {...busDetails} key={busDetails.busLine + i} />
            ))}
          </span>
        }
      />
    </ListItem>
  );
};

BusStopsListItem.propTypes = {
  name: proptypes.string.isRequired,
  id: proptypes.oneOfType([proptypes.string, proptypes.number]).isRequired,
  alternate: proptypes.bool.isRequired,
  buses: proptypes.arrayOf(
    proptypes.shape({
      busLine: proptypes.oneOfType([proptypes.string, proptypes.number]).isRequired,
      color: proptypes.string.isRequired,
    }),
  ).isRequired,
};

export default BusStopsListItem;
