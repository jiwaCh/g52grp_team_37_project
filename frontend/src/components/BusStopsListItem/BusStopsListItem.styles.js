import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => {
  const alternateColor = '#efefef';
  return {
    alternate: {
      color: theme.palette.getContrastText(alternateColor),
      backgroundColor: alternateColor,
    },
    chipRoot: {
      display: 'flex',
      justifyContent: 'left',
      flexWrap: 'wrap',
      '& > *': {
        margin: theme.spacing(0.5),
      },
    },
  };
});

export default useStyles;
