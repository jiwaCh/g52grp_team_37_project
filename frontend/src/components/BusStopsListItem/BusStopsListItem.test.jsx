import React from 'react';
import BusStopsListItem from './BusStopsListItem';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders correctly', () => {
  const props = {
    name: 'Some stop',
    id: '1234',
    alternate: false,
    buses: [],
  };

  const component = shallow(<BusStopsListItem {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});
