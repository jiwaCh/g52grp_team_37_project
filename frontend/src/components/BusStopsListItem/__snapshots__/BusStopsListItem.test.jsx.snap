// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(ListItem))
  button={true}
  className=""
  component={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "NavLink",
      "propTypes": Object {
        "activeClassName": [Function],
        "activeStyle": [Function],
        "aria-current": [Function],
        "className": [Function],
        "exact": [Function],
        "innerRef": [Function],
        "isActive": [Function],
        "location": [Function],
        "onClick": [Function],
        "replace": [Function],
        "strict": [Function],
        "style": [Function],
        "target": [Function],
        "to": [Function],
      },
      "render": [Function],
    }
  }
  key="1234"
  to="/busStop?stop=1234"
>
  <WithStyles(ForwardRef(ListItemIcon))>
    <LocationOnIcon />
  </WithStyles(ForwardRef(ListItemIcon))>
  <WithStyles(ForwardRef(ListItemText))
    primary="Some stop"
    secondary={
      <span
        className="makeStyles-chipRoot-2"
      />
    }
  />
</WithStyles(ForwardRef(ListItem))>
`;
